### ISSUE TITLE
Investigate if compressed request bodies affect CloudWatch Logs ingest pricing

### ISSUE THREAD

**Issue by schlarpc** [schlarpc/overengineered-cloudfront-s3-static-website#9]
Seems unlikely, but worth a look.

**Comment by schlarpc** [comment]
It doesn't appear to affect the IncomingBytes metric, so it's very unlikely to affect billing. Probably still worth implementing just for the network throughput savings though.

```python
def gzip_request_body(request, **_):
    if "Content-Encoding" not in request.headers:
        request.headers.add_header("Content-Encoding", "gzip")
        request.data = gzip.compress(request.body)

logs.meta.events.register("before-sign.logs.PutLogEvents", gzip_request_body)
```

**Comment by schlarpc** [comment]
Implemented in 66d11294e10bd87aaa191f7bbeb43b92c1e66c34
